File: src/services/aiEnhancement.ts
import { GoogleGenAI, Type } from "@google/genai";

const ai = new GoogleGenAI({ apiKey: process.env.GEMINI_API_KEY || "" });

/**
 * AI Enhancement service using Google Gemini.
 */
export class AiEnhancementService {
  /**
   * Extracts key skills from text.
   */
  async extractSkills(text: string): Promise<string[]> {
    if (!process.env.GEMINI_API_KEY) return [];
    
    try {
      const response = await ai.models.generateContent({
        model: "gemini-3-flash-preview",
        contents: `Extract a list of key professional skills from the following text. Return only the skills as a JSON array of strings.\n\nText: ${text.substring(0, 5000)}`,
        config: {
          responseMimeType: "application/json",
          responseSchema: {
            type: Type.ARRAY,
            items: { type: Type.STRING }
          }
        }
      });
      
      return JSON.parse(response.text || "[]");
    } catch (error) {
      console.error("AI Skill Extraction Error:", error);
      return [];
    }
  }

  /**
   * Generates a summary of a CV.
   */
  async summarizeCv(text: string): Promise<string> {
    if (!process.env.GEMINI_API_KEY) return "AI summary unavailable.";
    
    try {
      const response = await ai.models.generateContent({
        model: "gemini-3-flash-preview",
        contents: `Provide a concise 2-3 sentence professional summary of this candidate's CV.\n\nCV Text: ${text.substring(0, 5000)}`,
      });
      
      return response.text || "No summary generated.";
    } catch (error) {
      console.error("AI Summarization Error:", error);
      return "Error generating summary.";
    }
  }

  /**
   * Explains why a CV matches a job description.
   */
  async explainMatch(cvText: string, jdText: string): Promise<string> {
    if (!process.env.GEMINI_API_KEY) return "AI explanation unavailable.";
    
    try {
      const response = await ai.models.generateContent({
        model: "gemini-3-flash-preview",
        contents: `Explain why this candidate's CV matches the job description. Highlight specific matched skills and experiences. Keep it brief.\n\nCV: ${cvText.substring(0, 3000)}\n\nJob Description: ${jdText.substring(0, 3000)}`,
      });
      
      return response.text || "No explanation generated.";
    } catch (error) {
      console.error("AI Explanation Error:", error);
      return "Error generating explanation.";
    }
  }
}

File: python_implementation/ai_enhancement.py
import google.generativeai as genai
import os
import json

# Configure Gemini API
# genai.configure(api_key=os.getenv("GEMINI_API_KEY"))

def get_ai_summary(cv_text):
    """Generates a short summary of the candidate using AI."""
    model = genai.GenerativeModel('gemini-pro')
    prompt = f"Provide a concise professional summary of this candidate's CV:\n\n{cv_text[:4000]}"
    try:
        response = model.generate_content(prompt)
        return response.text
    except Exception as e:
        return f"AI Summary Error: {str(e)}"

def extract_skills_ai(text):
    """Extracts key skills from text using AI."""
    model = genai.GenerativeModel('gemini-pro')
    prompt = f"Extract key professional skills from this text. Return as a comma-separated list:\n\n{text[:4000]}"
    try:
        response = model.generate_content(prompt)
        return response.text
    except Exception as e:
        return f"AI Skill Extraction Error: {str(e)}"

def explain_match_ai(cv_text, jd_text):
    """Explains why a CV matches a job description using AI."""
    model = genai.GenerativeModel('gemini-pro')
    prompt = f"Explain why this CV matches the job description. Highlight specific matched skills:\n\nCV: {cv_text[:2000]}\n\nJD: {jd_text[:2000]}"
    try:
        response = model.generate_content(prompt)
        return response.text
    except Exception as e:
        return f"AI Explanation Error: {str(e)}"
